syntax = "proto2";
package protos;

import "easy_rec/python/protos/dnn.proto";


message DropoutNet {
  required DNN user_content = 1;
  required DNN user_preference = 2;
  required DNN item_content = 3;
  required DNN item_preference = 4;
  required DNN user_tower = 5;
  required DNN item_tower = 6;
  required float l2_regularization = 7 [default = 1e-4];
  optional float user_dropout_rate = 8 [default = 0];
  optional float item_dropout_rate = 9 [default = 0.5];
  required uint32 num_negative_mining = 10 [default = 4];
  optional string sample_weight_field = 11;
}
